bacs/src/main/java/com/adyen/checkout/bacs/DefaultBacsDirectDebitDelegate.kt

```diff
@@ -8,25 +8,61 @@
 
 package com.adyen.checkout.bacs
 
+import androidx.annotation.VisibleForTesting
 import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
 import com.adyen.checkout.components.model.payments.request.BacsDirectDebitPaymentMethod
 import com.adyen.checkout.components.model.payments.request.PaymentComponentData
+import com.adyen.checkout.components.ui.ViewProvider
+import com.adyen.checkout.components.ui.view.ComponentViewType
 import com.adyen.checkout.components.util.PaymentMethodTypes
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
 
 internal class DefaultBacsDirectDebitDelegate(
+    override val configuration: BacsDirectDebitConfiguration,
     val paymentMethod: PaymentMethod,
 ) : BacsDirectDebitDelegate {
 
+    override val inputData: BacsDirectDebitInputData = BacsDirectDebitInputData()
+
     private val _outputDataFlow = MutableStateFlow<BacsDirectDebitOutputData?>(null)
     override val outputDataFlow: Flow<BacsDirectDebitOutputData?> = _outputDataFlow
 
+    override val outputData
+        get() = _outputDataFlow.value
+
     private val _componentStateFlow = MutableStateFlow<BacsDirectDebitComponentState?>(null)
     override val componentStateFlow: Flow<BacsDirectDebitComponentState?> = _componentStateFlow
 
+    @VisibleForTesting
+    @Suppress("VariableNaming")
```

Comment:
Just curious: why did you have to suppress this? We use this naming convention everywhere 🤔

Reply:
It doesn't like this naming when the variable is not private :/

Reply:
Hmm I get why you did it like this, because it makes testing easier of course. But have you considered just calling `onInputDataChanged` and make this `private`? Or does that have too many side effects?

Reply:
I need to setup `_viewFlow` with a certain value before the test, but to do that with only `onInputDataChanged` means i need to call it with very specific values that will make the test confusing.